app/javascript/components/upload/ClusteringFileForm.jsx

```diff
@@ -20,20 +23,63 @@ export default function ClusteringFileForm({
   associatedClusterFileOptions=[],
   updateCorrespondingClusters,
   bucketName,
-  isInitiallyExpanded
+  isInitiallyExpanded,
+  isAnnDataExperience
 }) {
   const spatialClusterAssocs = file.spatial_cluster_associations
     .map(id => associatedClusterFileOptions.find(opt => opt.value === id))
   const validationMessages = validateFile({
     file, allFiles, allowedFileExts, requiredFields
   })
 
+
+  /** create the tooltip and message for the .obsm key names section */
+  function obsmKeyNameMessage() {
+    const obsmKeyNameToolTip = <span>
+      <OverlayTrigger
+        trigger={['hover', 'focus']}
+        rootClose placement="top"
+        overlay={obsmKeyNameHelpContent()}>
+        <span> .obsm key names <FontAwesomeIcon icon={faQuestionCircle}/></span>
+      </OverlayTrigger>
+    </span>
+
+    return <span >
+      {obsmKeyNameToolTip}
+    </span>
+  }
+
+  /** gets the popup message to describe .obsm keys */
+  function obsmKeyNameHelpContent() {
+    return <Popover id="cluster-obsm-key-name-popover" className="tooltip-wide">
+      <div> Multi-dimensional observations annotations .obsm (attribute) keys names for clusterings </div>
+    </Popover>
+  }
+
+  /**
+   * Configure the appropriate name form fields for Traditional or AnnData upload experience
+   */
+  function nameFields(isAnnDataExperience) {
+    if (isAnnDataExperience) {
+      return <div className="row">
+        <div className="col-md-6">
+          <TextFormField label="Name" fieldName="name" file={file} updateFile={updateFile}/>
+        </div>
+        <div className="col-md-6">
+          <TextFormField label= {obsmKeyNameMessage()} fieldName="obsm_key_names" file={file} updateFile={updateFile} placeholderText="x_tsne.."/>
```

Comment:
For `placeholder` we've used the E.g. format (like for gene search in studies):
```suggestion
          <TextFormField label= {obsmKeyNameMessage()} fieldName="obsm_key_names" file={file} updateFile={updateFile} placeholderText="E.g: 'x_tsne, x_umap'"/>
```

Reply:
Okay that looks nice I will make that update